src/ralph/virtual/models.py

```diff
@@ -258,6 +264,13 @@ class VirtualServer(AdminAbsoluteUrlMixin, NetworkableBaseObject, BaseObject):
     # TODO: remove this field
     cluster = models.ForeignKey(Cluster, blank=True, null=True)
 
+    def get_location(self):
+        return self.parent.get_location() if self.parent else None
```

Comment:
for virtual server, its location should be: `DC1 / SR1 / Rack 101 / 10 / s12345.mydc.net` - the last part is hostname of its supervisor.